### PR TITLE
Minor: do not crash if running_jobs is empty in restart file

### PR DESCRIPTION
**Pull request by oscarwumit**
Fixed an issue that crashes ARC during restart.

Previously, ARC crashes if the 'running_jobs' dictionary is empty in the restart file. This crash was found by Jim. Normally 'running_jobs' should not be empty in the restart file. However, if the user would like to restart ARC by ignoring previously running jobs, they might manually delete related job labels in the restart file. The user may forget to completely delete the 'running_jobs' key in the restart file, and left it with empty value.

This PR fixed the crash by assuming if 'running_jobs' is empty or None, then all jobs should have finished.

### CODE REVIEW

File: arc/scheduler.py [hunk 1]
```diff
@@ -2535,47 +2535,54 @@ def restore_running_jobs(self):
...
-                else:
-                    raise SchedulerError('Could not find species {0} in the restart file'.format(spc_label))
-                job = Job(job_dict=job_description)
-                if spc_label not in self.job_dict:
-                    self.job_dict[spc_label] = dict()
-                if job_description['job_type'] not in self.job_dict[spc_label]:
+        if not jobs:
+            del self.restart_dict['running_jobs']
+            self.running_jobs = dict()
+            logger.info('It seems that there are no running jobs specified in the ARC restart file. Assume all jobs'
```
Comment (alongd) [inline arc/scheduler.py]: Add a blank space after `Assume all jobs` (so it won't print `Assume all jobshave finished.`)
Also please change `Assume` to `Assuming`.

File: arc/scheduler.py [hunk 2]
```diff
@@ -2535,47 +2535,54 @@ def restore_running_jobs(self):
...
+                        'have finished.')
+        else:
+            logger.info(f'ARC restart files indicate the following jobs are still running. {jobs.keys()}')
```
Comment (alongd) [inline arc/scheduler.py]: Change to `f"ARC's restart file indicates that the  following"`...

**Comment by oscarwumit** [comment]
Thanks for the comment. Please take a look again. @alongd

File: arc/scheduler.py [hunk 3]
```diff
@@ -2545,47 +2545,54 @@ def restore_running_jobs(self):
...
-                if spc_label not in self.job_dict:
-                    self.job_dict[spc_label] = dict()
-                if job_description['job_type'] not in self.job_dict[spc_label]:
+        if not jobs:
+            del self.restart_dict['running_jobs']
+            self.running_jobs = dict()
+            logger.info('It seems that there are no running jobs specified in the ARC restart file. Assuming all jobs '
+                        'have finished.')
+        else:
+            logger.info(f"ARC's restart files indicate the following jobs are still running. {jobs.keys()}")
```
Comment (alongd) [inline arc/scheduler.py]: I think `job.keys()` returns an iterator, maybe we'd like to print `list(job.keys())` instead?

Comment (oscarwumit) [reply arc/scheduler.py]: Thanks @alongd

**Pull request merged**